nginx-1.0.6/objs/nginx.8
.\"
.\" Copyright (c) 2010 Sergey A. Osokin
.\" All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"
.Dd November 14, 2010
.Dt NGINX 8
.Os
.Sh NAME
.Nm nginx
.Nd "HTTP and reverse proxy server, mail proxy server"
.Sh SYNOPSIS
.Nm
.Op Fl hqtvV?
.Op Fl c Ar file
.Op Fl g Ar directives
.Op Fl p Ar prefix
.Op Fl s Ar signal
.Sh DESCRIPTION
The
.Nm
(spelled
.Dq engine x )
is an HTTP and reverse proxy server, as well as a mail proxy server.
The
.Nm
is known for its high performance, stability, rich feature set, simple
configuration, and low resource consumption.
.Pp
The options are as follows:
.Bl -tag -width ".Fl d Ar directives"
.It Fl ?\& | h
Print help.
.It Fl c Ar file
Use an alternative configuration
.Ar file .
.It Fl g Ar directives
Set global configuration directives.
See
.Sx EXAMPLES
for details.
.It Fl p Ar prefix
Set prefix path.
Default value is
.Pa /usr/local/nginx .
.It Fl q
Suppress non-error messages during configuration testing.
.It Fl s Ar signal
Send signal to the master process.
The argument
.Ar signal
can be one of:
.Cm stop , quit , reopen , reload .
The following table shows the corresponding system signals.
.Pp
.Bl -tag -width ".It Cm reopen" -compact
.It Cm stop
.Dv SIGTERM
.It Cm quit
.Dv SIGQUIT
.It Cm reopen
.Dv SIGUSR1
.It Cm reload
.Dv SIGHUP
.El
.It Fl t
Don't run, just test the configuration file.
The
.Nm
checks configuration for correct syntax and then tries to open files
referred in configuration.
.It Fl v
Print
.Nm
version.
.It Fl V
Print
.Nm
version, compiler version and
.Pa configure
script parameters.
.El
.Sh SIGNALS
The master process of
.Nm
can handle the following signals.
.Pp
.Bl -tag -width ".It Dv SIGINT , SIGTERM" -compact
.It Dv SIGINT , SIGTERM
Shut down quickly.
.It Dv SIGHUP
Reload configuration, start the new worker process with a new
configuration, gracefully shut down old worker processes.
.It Dv SIGQUIT
Shut down gracefully.
.It Dv SIGUSR1
Reopen log files.
.It Dv SIGUSR2
Upgrade
.Nm
executable on the fly.
.It Dv SIGWINCH
Shut down gracefully worker processes.
.El
.Pp
While there's no need to explicitly control worker processes normally,
they support some signals, too:
.Pp
.Bl -tag -width ".It Dv SIGINT , SIGTERM" -compact
.It Dv SIGTERM
Shut down quickly.
.It Dv SIGQUIT
Shut down gracefully.
.It Dv SIGUSR1
Reopen log files.
.El
.Sh DEBUGGING LOG
To enable a debugging log, reconfigure
.Nm
to build with debugging:
.Pp
.Dl "./configure --with-debug ..."
.Pp
and then set the
.Cm debug
level of the
.Va error_log :
.Pp
.Dl "error_log /path/to/log debug;"
.Pp
It is also possible to enable the debugging for some IP address:
.Bd -literal -offset indent
events {
	debug_connection 127.0.0.1;
}
.Ed
.Sh FILES
.Bl -tag -width indent -compact
.It Pa /var/run/nginx.pid
Contains the process ID of the
.Nm
listening for connections.
The content of this file is not sensitive; it can be world-readable.
.It Pa /etc/nginx/nginx.conf
Main configuration file.
.It Pa /var/log/nginx/error.log
Error log file.
.El
.Sh EXIT STATUS
Exit status is 0 on success, or 1 if the command fails.
.Sh EXAMPLES
.Bd -literal
nginx -t -c ~/mynginx.conf -g "pid /var/run/mynginx.pid; worker_processes 2;"
.Ed
Test configuration file
.Pa ~/mynginx.conf
with global directives for PID and quantity of worker processes.
.Sh SEE ALSO
.Xr nginx.conf 5
.Sh HISTORY
Development of
.Nm
started in 2002, with the first public release on October 4, 2004.
.Sh AUTHORS
.An Igor Sysoev Aq igor@sysoev.ru
.Pp
Documentation available on
.Pa http://nginx.org/
and
.Pa http://sysoev.ru/nginx/ .
.Pp
This manual page was written by
.An Sergey A. Osokin Aq osa@FreeBSD.org.ru
as a result of compilation of many
.Nm
documents all over the world.
.Sh BUGS
Report to mailing list
.Aq Li nginx@nginx.org
if you found one.


cpnginx/nginx.php
<?php
######################################################################################
#  Copyright (C) 2012 NginxCP.com. All rights reserved.
#
#  This program is free software; you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation; either version 2 of the License, or
#  (at your option) any later version.
#
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
######################################################################################

header("Content-Type: text/html\n\n");

// *** Common variables ***
$cpAppName = 'Nginx Admin';
$cpAppVersion = '5.2 Stable';
$NGINX_VERSION= str_replace('nginx version: nginx/','',shell_exec('nginx -v 2>&1'));

$user = getenv('REMOTE_USER');
if($user != "root") { echo "You do not have the proper permissions to access Nginx Admin..."; exit; }
function ejecutar($act) {
if($act == "restart") {
$var = shell_exec("/etc/init.d/httpd restart");
if(empty($var)) { echo "<p>Nginx Restarted Successfully.</p>"; } else { echo "<p>{$var}</p>"; }
} elseif($act == "rebuild") {
$var = shell_exec("/scripts/rebuildvhosts");
if(empty($var)) { echo "<p>Command running in background.</p>"; } else { echo "<p>{$var}</p>"; }
} elseif($act == "cleanup") {
		$var = shell_exec("rm -rvf /tmp/nginx_client/*");
		if(empty($var)) { echo "<p>No output generated. The folder /tmp/nginx_client is either empty or too full to be deleted using the \"rm\" command. Verify by checking the disk usage of the /tmp partition in WHM. If /tmp is over 70-80% used, you may want to try the following commands via SSH:<br /><br /># cd /tmp/nginx_client/<br /># find . -name '*' | xargs rm</p>"; } else { echo "<p>".nl2br($var)."</p>"; }
}
}
?>
<!DOCTYPE html>
<html>
<head>
<title><?php echo $cpAppName; ?></title>
<meta name="description" content="WHM Plug-in of Nginx for cPanel servers" />
<link rel='stylesheet' type='text/css' href='/themes/x/style_optimized.css' />
<script type="text/javascript">
function okay() {
if(confirm('Are you sure of save configuration?')) {
document.getElementById('okay').submit();
}
return false;
}
function clog() {
document.getElementById('log').submit();
}
</script>
<style>
div#wrap {
margin: 0 auto;
width: 700px;
}
</style>
</head>
<body class="yui-skin-sam">
<div id="pageheader">
        <div id="breadcrumbs">
                <p>&nbsp;<a href="/scripts/command?PFILE=main">Main</a> &gt;&gt; <a href="nginx.php" class="active"><?php echo 
$cpAppName; ?></a></p>
        </div>
<div id="doctitle"><h1><?php echo $cpAppName; ?> (v<?php echo $cpAppVersion; ?>) </h1><h4><span>(Nginx Version: (<?php echo $NGINX_VERSION; ?>)</span></h4>
</div>
<div id="wrap">
<table cellpadding="0" cellspacing="0">
<tr align="center">
<td width="180"><img src="restartservices.gif" alt="Restart Nginx" /></td>
<td width="180"><img src="config.png" alt="Nginx Configuration Editor" /></td>
<td width="180"><img src="rebuild.png" alt="Rebuild VHosts" /></td>
<td width="180"><img src="log.png" alt="View Nginx Log" /></td>
<td width="180"><img src="cleanup.gif" alt="cleanup" /></td>
</tr>
<tr align="center">
<td><a href="nginx.php?op=restart">Restart Nginx</a></td>
<td><a href="nginx.php?op=edit">Configuration Editor</a></td>
<td><a href="nginx.php?op=rebuild">Rebuild Vhosts</a></td>
<td><form action="nginx.php?op=logs" method="post" id="log">View the last?<br /><input type="text" name="l" value="25" autocomplete="off" /><br /><a href="nginx.php?op=logs" onClick="clog();return false;">View Log</a></form></td>
<td><a href="nginx.php?op=cleanup">Cleanup</a></td>
</tr>
</table><br />
<?php
$op = &$_GET['op'];
switch($op) {
case "restart": echo "<p style=\"color: #009\"><b>Restarting Nginx...</b></p>"; ejecutar("restart"); echo "<p style=\"color: #009\"><b>Done...</b></p>"; break;
case "edit":
if(isset($_POST['conf'])) {
$conf = $_POST['conf'];
file_put_contents("/etc/nginx/nginx.conf", $conf);
echo "<p><b>Configuration has been updated.</b></p>";
if(isset($_POST['c'])) { ejecutar("restart"); }
}
?>
<form action="nginx.php?op=edit" method="post" id="okay"><textarea name="conf" cols="80" rows="20"><?=shell_exec("cat /etc/nginx/nginx.conf")?></textarea><br />Restart Nginx? <input type="checkbox" name="c" /><br /><br /><input type="submit" value="Update!" onClick="okay();return false;" /></form>
<?
break;
case "rebuild": ejecutar("rebuild"); break;
case "logs":
if(empty($_POST['l'])) {
$var = shell_exec("cat /var/log/nginx/error.log");
$l = null;
} else {
$l = ereg_replace("/[0-9]/","",$_POST['l']);
$var = shell_exec("tail -{$l} /var/log/nginx/error.log");
}
echo "<p style=\"color: #009\"><b>Log Viewer - Showing Last {$l} Lines</b></p>";
echo "<pre>{$var}</pre>";
break;
case "cleanup": 
echo "<p style=\"color: #009\"><b>Temperary Files Cleanup:</b></p>";
ejecutar("cleanup"); break;
default:
$run = "Down";
$var = shell_exec("ps -A");
if(strstr($var,"nginx")) { $run = "UP"; }
echo "Nginx Service Status: <font style=\"color: #0c0\"><b>{$run}</b></font>";
?>
<p style="color: #0c0">To automated /tmp cleanup add bellow cron <br><b>0 */1 * * * /usr/sbin/tmpwatch -am 1 /tmp/nginx_client</b><br> via crontab -e command</p>
<p style="color: #03F"><b>About Nginx Admin</b></p>
<p>Nginx Admin is WHM interface of Nginx installer for cPanel server. This plugin will increase your server performance and decrease the server Apache Load. So you can host more websites in a cPanel server than usual. <br>Nginx is known for its high performance, stability, rich feature set, simple configuration, and low resource consumption. <br> Unlike traditional servers, Nginx doesn't rely on threads to handle requests. Instead it uses a much more scalable event-driven architecture. This architecture uses small, but more importantly, predictable amounts of memory under load. <br></p>
<p>For Support Visit <a href="http://www.nginxcp.com/forums" target='_blank'>http://www.nginxcp.com/forums</a></p>
<? } ?>

<!-- Begin PayPal Donations by http://johansteen.se/ -->
<form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_blank">
    <div class="paypal-donations">
        <input type="hidden" name="cmd" value="_donations" />
        <input type="hidden" name="business" value="9xlinux@live.com" />
        <input type="hidden" name="item_name" value="Nginx-Admin Open Source Project" />
        <input type="hidden" name="rm" value="0" />
        <input type="hidden" name="currency_code" value="USD" />
        <input type="image" src="https://www.paypal.com/en_US/i/btn/btn_donateCC_LG.gif" name="submit" alt="PayPal - The safer, easier way to pay online." />
        <img alt="" src="https://www.paypal.com/en_US/i/scr/pixel.gif" width="1" height="1" />
    </div>
</form>
<!-- End PayPal Donations -->

<p>Nginx Admin: v<?php echo $cpAppVersion; ?> <br>Nginx version: (<?php echo $NGINX_VERSION; ?>)</p><p>©2006-2014, <a href="http://www.nginxcp.com" 
target='_blank'>NginxCP.com</a></p>
</div>
</body>
</html>
